### ISSUE TITLE
The config manipulation interface should use a system message bus helper and does not belong in infopipe

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2399]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2395
- Created at 2014-07-31 13:33:15 by [simo](https://pagure.io/user/simo)
- Closed as Invalid
- Assigned to [pbrezina](https://pagure.io/user/pbrezina)
---

While working on the ticket to make sssd run as non root (#2370), it turned out interfaces to let openlmi write sssd.conf have been added to infopipe daemon.

This is the wrong place to put such interfaces, and they should be moved to a helper that is started on demand by the system message bus and is unrelated to the infopipe. A helper is fine as reconfigurations are rare and the overhead of running a helper is minimal.

This will allow to avoid needing root access from the infopipe to change the sssd.conf file (or changing the sssd.conf file ownerhip to let non-root sssd modify it), making non-root sssd better as it will not be able to change configurations diorectly (the helper will run as root).

### Comments

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-08-06 09:43:11](https://pagure.io/SSSD/sssd/issue/2395#241292)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.12.1

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-08-06 09:44:15](https://pagure.io/SSSD/sssd/issue/2395#241293)**

Fields changed

rhbz:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-09-08 20:08:46](https://pagure.io/SSSD/sssd/issue/2395#241294)**

Mass-moving all tickets that didn't make 1.12.1 into 1.12.2

milestone: SSSD 1.12.1 => SSSD 1.12.2

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-09-30 19:06:14](https://pagure.io/SSSD/sssd/issue/2395#241295)**

We need to do a release as requested by downstream. Moving tickets that are not fixed already or very close to acking to 1.12.3

milestone: SSSD 1.12.2 => SSSD 1.12.3

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-10-23 14:32:15](https://pagure.io/SSSD/sssd/issue/2395#241296)**

Fields changed

mark:  => 0
priority: critical => major

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-11-19 18:01:47](https://pagure.io/SSSD/sssd/issue/2395#241297)**

Fields changed

owner: somebody => pbrezina

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-11-19 18:03:08](https://pagure.io/SSSD/sssd/issue/2395#241298)**

We need to finish some rootless sssd tasks first, then Pavel will do an assesment of this ticket.

---

**Comment from [simo](https://pagure.io/user/simo) at [2014-11-27 20:13:19](https://pagure.io/SSSD/sssd/issue/2395#241299)**

Fields changed

milestone: SSSD 1.12.3 => SSSD 1.12.4

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-01-21 16:05:30](https://pagure.io/SSSD/sssd/issue/2395#241300)**

The sbus-related patches are landing on the list, but we realized during review that they're too big for 1.12, at least for the time being. We're going to commit them to master only for now and backport on request to avoid breaking 1.12.

milestone: SSSD 1.12.4 => Tools Deferred

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-10 17:55:11](https://pagure.io/SSSD/sssd/issue/2395#241301)**

Sorry, wrong milestone.

milestone: Tools Deferred => SSSD 1.13 alpha

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-02-12 11:05:33](https://pagure.io/SSSD/sssd/issue/2395#241302)**

Fields changed

summary: The openlmi interface should use a system message bus helper and does not belong in infopipe => The config manipulation interface should use a system message bus helper and does not belong in infopipe

---

**Comment from [simo](https://pagure.io/user/simo) at [2015-06-18 16:08:22](https://pagure.io/SSSD/sssd/issue/2395#241303)**

Fields changed

milestone: SSSD 1.13 alpha => SSSD 1.13 backlog
sensitive:  => 0

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-02-16 15:32:59](https://pagure.io/SSSD/sssd/issue/2395#241304)**

We should just rip out the config management API from SSSD completely..

milestone: SSSD 1.13 backlog => SSSD 1.14 alpha

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-02-16 17:24:00](https://pagure.io/SSSD/sssd/issue/2395#241305)**

Replying to [comment:13 jhrozek]:
> We should just rip out the config management API from SSSD completely..

Can you please elaborate?

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-02-16 17:52:14](https://pagure.io/SSSD/sssd/issue/2395#241306)**

Replying to [comment:14 dpal]:
> Replying to [comment:13 jhrozek]:
> > We should just rip out the config management API from SSSD completely..
> 
> Can you please elaborate?

We have a config API exposed on D-Bus. Because sssd.conf is currently writable as root only, this limits the IFP API to run as root as well. Now we have two options:
 1. leave the D-Bus configuration API around, but move it to another D-Bus service started on demand. This is what simo was proposing earlier
 2. Remove this D-Bus config API from SSSD completely.

Because AFAIK this API is only used by OpenLMI which is more or less dead, I was leaning towards 2. But whatever we do, I would really like the IFP responder to not run as root in 1.14. Implementing 1. would just be a bit more work and I'm not sure there would be any users.

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-02-16 17:59:54](https://pagure.io/SSSD/sssd/issue/2395#241307)**

I think we would still need the configuration API. Moving it to a service but not completing the work on that service until we need this API is fine but just ripping it out is IMO the wrong approach as this is how SSSD should be configured via tools like ansible and cockpit. I also suspect we will use it in future to manage SSSD configuration remotely via cockpit.

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-06-20 11:59:58](https://pagure.io/SSSD/sssd/issue/2395#241308)**

This needs a fair amount of work which is not in the scope of 1.14 unfortunately.

milestone: SSSD 1.14 alpha => SSSD 1.15 beta

---

**Comment from [simo](https://pagure.io/user/simo) at [2016-11-25 10:09:13](https://pagure.io/SSSD/sssd/issue/2395#241309)**

We actually removed the config manipulation completely.

resolution:  => wontfix
status: new => closed

---

**Comment from [simo](https://pagure.io/user/simo) at [2017-02-24 15:04:02](https://pagure.io/SSSD/sssd/issue/2395#241310)**

**Metadata Update from @simo**:
- Issue assigned to pbrezina
- Issue marked as blocked by: #2370
- Issue set to the milestone: SSSD Future releases (no date set yet)
